connect/connect.go

```diff
@@ -0,0 +1,292 @@
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf(err.Error())
+	}
+
+	//unmarshal response json and store in csrInfo
+	response, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		return nil, fmt.Errorf(err.Error())
+	}
+
+	type eventsInfoStruct struct {
+		EventsInfo []types.EventResponse `json:"eventsInfo"`
+	}
+
+	eventsInfoObj := eventsInfoStruct{}
+	err = json.Unmarshal(response, &eventsInfoObj)
+	return eventsInfoObj.EventsInfo, err
+}
+
+func GetKymaConnector(writerInterface WriterInterface) *KymaConnector {
+	c := &KymaConnector{
+		CsrInfo:          &types.CSRInfo{},
+		Ca:               &types.ClientCertificate{},
+		Info:             &types.Info{},
+		StorageInterface: writerInterface,
+	}
+
+	c.loadConfig()
+	c.populateClient()
+	return c
+}
+
+func (c *KymaConnector) GetSecureClient() (*http.Client, error) {
+	cert, err := tls.X509KeyPair([]byte(c.Ca.PublicKey), []byte(c.Ca.PrivateKey))
+	if err != nil {
+		return nil, fmt.Errorf(err.Error())
+	}
+
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+	}
+
+	tlsConfig.BuildNameToCertificate()
+	transport := &http.Transport{TLSClientConfig: tlsConfig}
+
+	return &http.Client{Transport: transport}, nil
+
+}
+
+func (c *KymaConnector) RenewCertificateSigningRequest() error {
+	keys, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	csr, err := c.getCsr(keys)
+	type csrStruct struct {
+		Csr string `json:"csr"`
+	}
+	encodedCsr := base64.StdEncoding.EncodeToString(csr)
+
+	requestBody := csrStruct{
+		Csr: encodedCsr,
+	}
+
+	jsonBytes, err := json.Marshal(requestBody)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	resp, err := c.SecureClient.Post(c.Info.URLs.RenewCertUrl, "application/json", bytes.NewBuffer(jsonBytes))
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusCreated {
+		log.Printf("Successfully renewed certificate")
+	} else {
+		return errors.New("Failed to renew certificate")
+	}
+
+	certificates, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+	crtResponse := types.CRTResponse{}
+	err = json.Unmarshal(certificates, &crtResponse)
+	if err != nil {
+		return fmt.Errorf("JSON Error")
+	}
+	decodedCert, err := base64.StdEncoding.DecodeString(crtResponse.ClientCRT)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	c.Ca.PublicKey = string(decodedCert)
+
+	c.StorageInterface.WriteClientCert(c.Ca)
+
+	return err
+}
+
+func (c *KymaConnector) RevokeCertificate() error {
+	resp, err := c.SecureClient.Post(c.Info.URLs.RevokeCertUrl, "application/json", nil)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusCreated {
+		log.Print("Successfully revoked certificate for client")
+	} else {
+		return errors.New("Error in trying to revoke certificate")
+	}
+	return err
```

Comment:
```suggestion
	return errors.New("Error in trying to revoke certificate")
```

Reply:
This is not gone work this way but we can simplify the code here.